src/astro/files/type/ndjson.py

```diff
@@ -0,0 +1,59 @@
+import io
+import json
+from typing import Union
+
+import pandas as pd
+
+from astro.constants import DEFAULT_CHUNK_SIZE
+from astro.constants import FileType as FileTypeConstants
+from astro.files.type.base import FileType
+
+
+class NDJSONFileType(FileType):
+    def read_to_dataframe(self, stream, **kwargs):
+        """read ndjson file from one of the supported locations and return dataframe
+
+        :param stream: file stream object
+        """
+        normalize_config = kwargs.pop("normalize_config")
+        return self.flatten_ndjson(normalize_config, stream)
+
+    def write_from_dataframe(self, df: pd.DataFrame, stream: io.TextIOWrapper) -> None:
+        """Write ndjson file to one of the supported locations
+
+        :param df: pandas dataframe
+        :param stream: file stream object
+        """
+        df.to_json(stream, orient="records", lines=True)
+
+    @property
+    def name(self):
+        return FileTypeConstants.NDJSON
+
+    def flatten_ndjson(
```

Comment:
Since we're within NDJSON, perhaps this method could be called `normalize` or `flatten`

Reply:
Done.